pkg/platform/openstack/kuryr_bootstrap.go

```diff
@@ -380,18 +381,23 @@ func BootstrapKuryr(conf *operv1.NetworkSpec, kubeClient client.Client) (*bootst
 
 	var sgRules = []sgRule{
 		{podSgId, "0.0.0.0/0", 0, 0, rules.ProtocolTCP},
-		{masterSgId, openStackSvcCIDR, etcdPort, etcdPort, rules.ProtocolTCP},
+		{masterSgId, openStackSvcCIDR, etcdClientPort, etcdClientPort, rules.ProtocolTCP},
 		{masterSgId, openStackSvcCIDR, apiPort, apiPort, rules.ProtocolTCP},
+		// NOTE (maysamacedo): Splitting etcd sg port ranges in different
+		// rules to avoid the issue of constant leader election changes
+		{masterSgId, workerSubnet.CIDR, etcdClientPort, etcdClientPort, rules.ProtocolTCP},
+		{masterSgId, workerSubnet.CIDR, etcdServerToServerPort, etcdServerToServerPort, rules.ProtocolTCP},
 	}
 
 	var decommissionedRules = []sgRule{
 		{podSgId, workerSubnet.CIDR, 0, 0, ""},
-		{masterSgId, openStackSvcCIDR, 2379, 2380, rules.ProtocolTCP},
+		{masterSgId, openStackSvcCIDR, etcdClientPort, etcdServerToServerPort, rules.ProtocolTCP},
+		{masterSgId, workerSubnet.CIDR, etcdClientPort, etcdServerToServerPort, rules.ProtocolTCP},
```

Comment:
Uhm, had this rule ever existed?

Reply:
The installer creates it. This is the one that needs to be deleted and splitted.

Reply:
Oh crap, so this is certainly a bigger problem. First of all it'll affect setups without Kuryr too, right? If so it probably should be changed in installer too.

Second issue is the fact that IIRC we're unable to modify those rules on upgrade, so older releases are stuck with old groups.